google/cloud/bigquery/client.py

```diff
@@ -1625,11 +1626,9 @@ def create_job(self, job_config, retry=DEFAULT_RETRY):
                 job_config
             )
             destination = _get_sub_prop(job_config, ["copy", "destinationTable"])
+            destination = TableReference.from_api_repr(destination)
```

Comment:
Would you like to guard this check with `isinstance(TableReference, destination)`?

Reply:
I mock further in test so now it cover `to_api_repr`.  I am not sure about the guard to check instance because `job_config` is dict and as per the example it has values in the string format.